internal/controller/controllers/common.go

```diff
@@ -268,6 +268,37 @@ func checksumMap(m map[string]string) (string, error) {
 	return fmt.Sprintf("%x", hash.Sum(nil)), nil
 }
 
+// checksumSecret produces a checksum of a Secret's Data attribute. The checksum
+// can be used to detect when the contents of a Secret have changed.
+func checksumSecret(m map[string][]byte) (string, error) {
```

Comment:
Looks like we should be able to deduplicate at least some of this and `checksumMap` given that we're treating the data as bytes in both